### PR TITLE
Include REST module's tests in GHA

### PR DESCRIPTION
**Pull request by PenghaiZhang**
#1480 ### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [ ] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change

The purpose of this PR is to include tests of module `oeq-ts-rest-api` in GHA.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -259,3 +259,78 @@ jobs:
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Install Chromium
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y --no-install-recommends \
+            chromium-browser \
+            chromium-chromedriver \
```
Comment (ChristianMurphy) [inline .github/workflows/ci.yaml:300]: FWIW the github linux virtual environment comes with Chrome + ChromeDriver and Firefox + GeckoDriver pre-installed https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#ubuntu-20041-lts

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -259,3 +259,78 @@ jobs:
...
+        shell: bash -l {0}
+        run: |
+          curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+          export NVM_DIR="$HOME/.nvm"
+          echo "source $NVM_DIR/nvm.sh" >> "$HOME/.bash_profile"
+
+      - name: Install NodeJS
+        shell: bash -l {0}
+        run: |
+          nvm install
```
Comment (ChristianMurphy) [inline .github/workflows/ci.yaml:331]: https://github.com/openequella/openEQUELLA/pull/2284 may work here as well

**Review by ChristianMurphy — COMMENTED** [review]
Thanks @PenghaiZhang!
It's good to see more tests making their way into CI! :+1:

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (PenghaiZhang) [reply .github/workflows/ci.yaml:331]: Thanks!!  I should action `setup-node`!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (PenghaiZhang) [reply .github/workflows/ci.yaml]: Nice! Thanks @ChristianMurphy !

Comment (ChristianMurphy) [reply .github/workflows/ci.yaml]: hmmm it looks like oEQ may be hard coded to look for chromium, even if chrome is present.

Comment (PenghaiZhang) [reply .github/workflows/ci.yaml]: Yes, in `github-actions.conf` we specify where to look for chromedriver. Do you know where is the pre-installed chromedriver ? I am about to do some search.

Comment (edalex-ian) [reply .github/workflows/ci.yaml]: It's in the autotest.conf. So if you change this, make sure you also change it for the functional test.

Comment (ChristianMurphy) [reply .github/workflows/ci.yaml]: > Do you know where is the pre-installed chromedriver ?

I'm not sure the exact location, but the docs mention

> Chrome Driver is available via CHROMEWEBDRIVER environment variable

so I'm guessing using that directly, or echo'ing it and the coding it in would work.

https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#ubuntu-20041-lts

Comment (PenghaiZhang) [reply .github/workflows/ci.yaml]: Yes I also found this. I will push a commit soon.

Comment (PenghaiZhang) [reply .github/workflows/ci.yaml]: https://github.com/actions/virtual-environments/issues/248

**Review by edalex-ian — COMMENTED** [review]
LGTM... mostly. :grin: 

Ideally it'd be great to see if we could create a custom action which did the four steps of setting up oEQ - i.e. setting up the DB, downloading the artefacts and setting up oEQ. We could put this in its own repo, and then later when we split this out we could share it. But, maybe that's a bit much for here...

... hmmm, actually, then again, those four steps are now duplicated so we should probably do something.. Either do a custom action just in this repo, or maybe use a YAML reference.

What do you think?

**Comment by PenghaiZhang** [comment]
> LGTM... mostly. 
> 
> Ideally it'd be great to see if we could create a custom action which did the four steps of setting up oEQ - i.e. setting up the DB, downloading the artefacts and setting up oEQ. We could put this in its own repo, and then later when we split this out we could share it. But, maybe that's a bit much for here...
> 
> ... hmmm, actually, then again, those four steps are now duplicated so we should probably do something.. Either do a custom action just in this repo, or maybe use a YAML reference.
> 
> What do you think?

I prefer to create a composite run steps action. Do you it's good to try ?

**Comment by edalex-ian** [comment]
Yup, definitely! That's what I meant by this clumsy bit of wording:

> custom action just in this repo

**Comment by PenghaiZhang** [comment]
> Yup, definitely! That's what I meant by this clumsy bit of wording:
> 
> > custom action just in this repo
> 
> Oh, other than it has to go in another repo... But ah well, that'll set us up for when we split out.

Yes! So would you like to create a new repo under `openequella`?

**Comment by edalex-ian** [comment]
Hmmm, I'm not sure it'll work.. Based on [a comment from a GH staffer](https://github.community/t/reusing-sharing-inheriting-steps-between-jobs-declarations/16851/17) you may only be able to use these for `run` actions:

> You can use composite run steps to make your run steps reusable across actions 101 (but not use actions within actions)

Maybe we're facing a fundamental DRY issue with GHA. But there's lots of discussion out there about them adding YAML Anchors etc, so maybe in the future we'll have better options. :disappointed: 

Perhaps we leave this (as is) for now. :thinking:

**Comment by edalex-ian** [comment]
So... Yeah, I reckon let's merge now. :+1:

**Pull request merged**